podman/domain/manifests.py

```diff
@@ -0,0 +1,247 @@
+        """Add Image to manifest list.
+
+        Args:
+            images: List of Images to be added to manifest.
+
+        Keyword Args:
+            all (bool):
+            annotation (Dict[str, str]):
+            arch (str):
+            features (List[str]):
+            os (str):
+            os_version (str):
+            variant (str):
+
+        Raises:
+            ImageNotFound: When Image(s) could not be found
+            APIError: When service reports an error
+        """
+        params = {
+            "all": kwargs.get("all"),
+            "annotation": kwargs.get("annotation"),
+            "arch": kwargs.get("arch"),
+            "features": kwargs.get("features"),
+            "images": list(),
+            "os": kwargs.get("os"),
+            "os_version": kwargs.get("os_version"),
+            "variant": kwargs.get("variant"),
+        }
+        for item in images:
+            if isinstance(item, Image):
+                item = item.attrs["RepoTags"][0]
+            params["images"].append(item)
+
+        data = api.prepare_body(params)
+        response = self.client.post(f"/manifests/{self.quoted_name}/add", data=data)
+
+        if response.status_code == requests.codes.okay:
+            return self.reload()
+
+        body = response.json()
+        if response.status_code == requests.codes.not_found:
+            raise ImageNotFound(body["cause"], response=response, explanation=body["message"])
+        raise APIError(body["cause"], response=response, explanation=body["message"])
+
+    def push(
+        self, destination: str, all: Optional[bool] = None  # pylint: disable=redefined-builtin
+    ) -> None:
+        """Push a manifest list or image index to a registry.
+
+        Args:
+            destination: Target for push.
+            all: Push all images.
+
+        Raises:
+            NotFound: When the Manifest could not be found
+            APIError: When service reports an error
+        """
+        params = {
+            "all": all,
+            "destination": destination,
+        }
+        response = self.client.post(f"/manifests/{self.quoted_name}/push", params=params)
+
+        if response.status_code == requests.codes.okay:
+            return
+
+        body = response.json()
+        if response.status_code == requests.codes.not_found:
+            raise NotFound(body["cause"], response=response, explanation=body["message"])
+        raise APIError(body["cause"], response=response, explanation=body["message"])
+
+    def remove(self, digest: str) -> None:
+        """Remove Image digest from manifest list.
+
+        Args:
+            digest: Image digest to be removed. Should a full Image reference be provided
+                the digest will be parsed out.
+
+        Raises:
+            ImageNotFound: When the Image could not be found
+            APIError: When service reports an error
+        """
+        if "@" in digest:
+            _, digest = digest.split("@")
```

Comment:
I realize having multiple `@` in the image unexpected, but I would encourage consideration of some defensive "input-validation" here. Consider what happens if someone hand-crafts an image for malicious purposes.  What about:

```
digest = digest.split("@", maxsplit=2)[1]
```